dash_auth/oidc_auth.py

```diff
@@ -0,0 +1,298 @@
+            Whether to force redirection to https, by default None
+            This is useful when the HTTPS termination is upstream of the server
+            If a string is passed, this will check for the existence of
+            an envvar with that name and force https callback if it exists.
+        client_kwargs : dict, optional
+            Keyword arguments passed to the OAuth client
+        login_route : str, optional
+            The route for the login function, it requires a <idp>
+            placeholder, by default "/oidc/<idp>/login".
+        logout_route : str, optional
+            The route for the logout function, by default "/oidc/logout".
+        callback_route : str, optional
+            The route for the OIDC redirect URI, it requires a <idp>
+            placeholder, by default "/oidc/<idp>/callback".
+        idp_selection_route : str, optional
+            The route for the IDP selection function, by default None
+        log_signins : bool, optional
+            Whether to log signins, by default False
+        **kwargs
+            Additional keyword arguments are passed to oauth.register.
+
+        Raises
+        ------
+        Exception
+            Raise an exception if the app.server.secret_key is not defined
+        """
+        super().__init__(app, public_routes=public_routes)
+        if force_https_callback is not None:
+            self.force_https_callback = (
+                os.getenv(force_https_callback) is not None
+                if isinstance(force_https_callback, str)
+                else force_https_callback
+            )
+        else:
+            self.force_https_callback = False
+
+        self.login_route = login_route
+        self.logout_route = logout_route
+        self.callback_route = callback_route
+        self.log_signins = log_signins
+        self.idp_selection_route = idp_selection_route
+        self.logout_page = logout_page
+
+        if secret_key is not None:
+            app.server.secret_key = secret_key
+
+        if app.server.secret_key is None:
+            raise RuntimeError(
+                """
+                app.server.secret_key is missing.
+                Generate a secret key in your Python session
+                with the following commands:
+                >>> import os
+                >>> import base64
+                >>> base64.b64encode(os.urandom(30)).decode('utf-8')
+                and assign it to the property app.server.secret_key
+                (where app is your dash app instance), or pass is as
+                the secret_key argument to OIDCAuth.__init__.
+                Note that you should not do this dynamically:
+                you should create a key and then assign the value of
+                that key in your code/via a secret.
+                """
+            )
+
+        super().__init__(app)
+
+        self.oauth = OAuth(app.server)
+
+        # Check that the login and callback rules have an <idp> placeholder
```

Comment:
do we also need to ensure idp follow a few rules ( no spaces, no caps...)?

Reply:
It just needs to not have slashes. Added that check in the register_provider.